src/server.ts

```diff
@@ -87,9 +93,15 @@ export class Server extends EventEmitter {
    * End all connections and disconnect the plugin
    */
   async close (): Promise<void> {
+    // Stop handling new requests: close all streams and the connection
     await this.pool.close()
     this.plugin.deregisterDataHandler()
+
+    // Wait for in-progress requests to finish
+    await this.pendingRequests
+    await new Promise(r => setTimeout(r, 100))
```

Comment:
let's make this configurable instead of a magic number. People might want different values depending on latency and such